metrics/metrics.go

```diff
@@ -54,6 +54,36 @@ var (
 		},
 	)
 
+	// K8sAPIDupReqCount tracks total number of K8s Api requests performed when duplicated pods are identified in the cache.
+	K8sAPIDupReqCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: "iam",
+			Name:      "k8s_dup_req_count",
+			Help:      "Total number of K8s Api requests performed when duplicated pods are identified in the cache.",
+		},
+	)
+
+	// K8sAPIDupReqSuccesCount tracks total number of times we successfully retrieve the pod from the K8s Api.
+	K8sAPIDupReqSuccesCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: "iam",
+			Name:      "k8s_dup_req_success_count",
+			Help:      "Total number of times we successfully retrieve the pod from the K8s Api.",
+		},
+	)
+
+	// PodNotFoundInCache tracks total number of times we don't have the pod info in the cache.
+	PodNotFoundInCache = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: "iam",
+			Name:      "k8s_pod_not_in_pod_cache",
```

Comment:
`s/k8s_pod_not_in_pod_cache/k8s_indexer_pod_not_found/g`.  We don't know if the indexer is using a cache or not, just that the indexer returned no results.